cmd/ingress/main.go

```diff
@@ -19,62 +19,132 @@ package main
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"log"
-	"os"
+	"net/http"
 
 	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/kelseyhightower/envconfig"
 	"github.com/streadway/amqp"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/pkg/logging"
 )
 
-var (
-	brokerURL    = os.Getenv("BROKER_URL")
-	exchangeName = os.Getenv("EXCHANGE_NAME")
+const (
+	defaultMaxIdleConnections        = 1000
+	defaultMaxIdleConnectionsPerHost = 1000
 )
 
+type envConfig struct {
+	Port         int    `envconfig:"PORT" default:"8080"`
+	BrokerURL    string `envconfig:"BROKER_URL" required:"true"`
+	ExchangeName string `envconfig:"EXCHANGE_NAME" required:"true"`
+
+	channel *amqp.Channel
```

Comment:
Just curious on why to shove these into envConfig instead of keeping the separation here between incoming environmental variables.

Reply:
because I needed an instance of a struct to wire the send function and ServeHTTP to and all this code is local to the main file here. The right way would be to move this into a pkg folder somewhere I think

Reply:
Right, that's what I was thinking as well (move to pkg) so if it's not in the envConfig now might make it easier.